src/dotnet/Contacts.Service/ContactsMigrationBackend.cs

```diff
@@ -0,0 +1,68 @@
+using ActualChat.Contacts.Db;
+using Microsoft.EntityFrameworkCore;
+using Stl.Fusion.EntityFramework;
+
+namespace ActualChat.Contacts;
+
+public class ContactsMigrationBackend(IServiceProvider services) : DbServiceBase<ContactsDbContext>(services), IContactsMigrationBackend
+{
+    private IContactsBackend ContactsBackend { get; } = services.GetRequiredService<IContactsBackend>();
+
+    public virtual async Task OnMoveChatToPlace(ContactsMigrationBackend_MoveChatToPlace command, CancellationToken cancellationToken)
+    {
+        var (chatId, placeId) = command;
+        var placeChatId = new PlaceChatId(PlaceChatId.Format(placeId, chatId.Id));
+        var newChatId = (ChatId)placeChatId;
+
+        if (Computed.IsInvalidating()) {
+            if (ContactsBackend is ContactsBackend contactsBackend) {
+                _ = contactsBackend.Internals.PseudoChatContact(chatId);
+                _ = contactsBackend.Internals.PseudoChatContact(newChatId);
+                _ = contactsBackend.Internals.PseudoPlaceContact(placeId);
+            }
+            return;
+        }
+
+        var chatSid = chatId.Value;
+        var placeSid = placeId.Value;
+
+        var dbContext = await CreateCommandDbContext(cancellationToken).ConfigureAwait(false);
+        await using var __ = dbContext.ConfigureAwait(false);
+
+        var dbContacts = await dbContext.Contacts
+            .Where(c => c.ChatId == chatSid)
+            .ToListAsync(cancellationToken)
+            .ConfigureAwait(false);
+
+        // Update DbContacts
+        foreach (var dbContact in dbContacts) {
+            var contactSid = dbContact.Id;
+            var ownerId = new UserId(dbContact.OwnerId);
+            var newContactSid = ContactId.Format(ownerId, newChatId);
+            var newChatSid = newChatId.Value;
+
+            var updateCount = await dbContext.Contacts
+                .Where(c => c.Id == contactSid)
+                .ExecuteUpdateAsync(setters => setters
+                        .SetProperty(c => c.Id, c => newContactSid)
+                        .SetProperty(c => c.ChatId, c => newChatSid)
+                        .SetProperty(c => c.PlaceId, c => placeSid),
+                    cancellationToken)
+                .ConfigureAwait(false);
+            AssertOneUpdated(updateCount);
+        }
+
+        await dbContext.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
+
+        // Place contacts should be added on adding authors to place root chat during processing ChatsMigrationBackend_MoveToPlace.
+    }
+
+    private static void AssertOneUpdated(int updateCount)
```

Comment:
I'd move all these methods to `ActualChat.Db` and implement them as extensions to `Task<int>` - `RequireUpdated`, `RequireOneUpdated`, `RequireAtLeastOneUpdated`.

Reply:
Done